karapace/avro_compatibility.py

```diff
@@ -28,12 +30,24 @@ def parse_json_ignore_trailing(s: str) -> Any:
     return SchemaFromJSONData(json_data, names)
 
 
+def is_incompatible(result: "SchemaCompatibilityResult") -> bool:
+    return result.compatibility is SchemaCompatibilityType.incompatible
+
+
+def is_compatible(result: "SchemaCompatibilityResult") -> bool:
```

Comment:
Do you actually need both? Does is_compatible != not is_incompatible?

Reply:
yes, there are more than two states:

https://github.com/aiven/karapace/blob/4737f82cb685444aaa49d6aabb5edc0a6e0392ea/karapace/avro_compatibility.py#L31-L34

I'm not sure how important the state `recursion_in_progress` is for the Avro compatibility check, so I have not changed it.

Reply:
That is unusual. I would say that recursion_in_progress is another type, that just wrapped around compatibility type. But it is not important right now.